### PR TITLE
db.ping() must use the same application level query timing and reconnect...

### PR DESCRIPTION
**Pull request by lloyd**
...ion logic as other query types - issue #1608

### CODE REVIEW

**Comment by lloyd** [comment]
NOTE: if you were to diff -w this, it would not look nearly as scary.  I suggest this one for immediate merge into dev because it's a 5 star that exacerbated a production outage, and is low riskish.

File: lib/db/mysql_wrapper.js [hunk 1]
```diff
@@ -39,14 +39,15 @@ exports.createClient = function(options) {
...
-          cb("database is intentionally stalled");
-        });
-      } else {
-        this.realClient.ping(cb);
-      }
+    ping: function(client_cb) {
+      // ping queries are added to the front of the pending work queue.  they are
+      // a priority, as they are used by load balancers that want to know the health
+      // of the system.
+      queryQueue.unshift({
```
Comment (ozten) [inline lib/db/mysql_wrapper.js:46]: Very cool. This makes sense.

**Comment by jedp** [comment]
regarding diff -w, i just learned that you can add ?w=1 to the diff url and it ignores whitespace

**Comment by ozten** [comment]
Great work getting actual STR by interviewing pete.
r+

**Pull request merged**